cannot install hmmleran in `python:3.11` image

I tried to install hmmlearn in the official Python 3.11 image `python:3.11`, but got errors.

```sh
docker run -it python:3.11 bash
pip install -v hmmlearn
```
Logs:

```
docker run -it python:3.11 bash
Unable to find image 'python:3.11' locally
3.11: Pulling from library/python
32fb02163b6b: Pull complete 
167c7feebee8: Pull complete 
d6dfff1f6f3d: Pull complete 
e9cdcd4942eb: Pull complete 
ca3bce705f6c: Pull complete 
5e1c6c4f8bbf: Pull complete 
2da42ff3382c: Pull complete 
86f9457966ab: Pull complete 
896264e2a03c: Pull complete 
Digest: sha256:0d753a7365274cef746b34dde9b4aaa27644f64e1567ed8f40ccd191ac4bd530
Status: Downloaded newer image for python:3.11
root@eba293a3a664:/# pip install -v hmmlearn
Using pip 22.3.1 from /usr/local/lib/python3.11/site-packages/pip (python 3.11)
Collecting hmmlearn
  Downloading hmmlearn-0.2.8.tar.gz (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.9/61.9 kB 9.3 MB/s eta 0:00:00
  Running command python setup.py egg_info
  /usr/local/lib/python3.11/site-packages/setuptools/config/setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
    warnings.warn(msg, warning_class)
  /usr/local/lib/python3.11/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  running egg_info
  creating /tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info
  writing /tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/dependency_links.txt
  writing requirements to /tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/requires.txt
  writing top-level names to /tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/top_level.txt
  writing manifest file '/tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/SOURCES.txt'
  listing git files failed - pretending there aren't any
  reading manifest file '/tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/SOURCES.txt'
  adding license file 'LICENSE.txt'
  writing manifest file '/tmp/pip-pip-egg-info-t15kztvu/hmmlearn.egg-info/SOURCES.txt'
  Preparing metadata (setup.py) ... done
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/3a/be/650f9c091ef71cb01d735775d554e068752d3ff63d7943b26316dc401749/numpy-1.21.2.zip (from https://pypi.org/simple/numpy/) (requires-python:>=3.7,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/5f/d6/ad58ded26556eaeaa8c971e08b6466f17c4ac4d786cd3d800e26ce59cc01/numpy-1.21.3.zip (from https://pypi.org/simple/numpy/) (requires-python:>=3.7,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/fb/48/b0708ebd7718a8933f0d3937513ef8ef2f4f04529f1f66ca86d873043921/numpy-1.21.4.zip (from https://pypi.org/simple/numpy/) (requires-python:>=3.7,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/c2/a8/a924a09492bdfee8c2ec3094d0a13f2799800b4fdc9c890738aeeb12c72e/numpy-1.21.5.zip (from https://pypi.org/simple/numpy/) (requires-python:>=3.7,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/45/b7/de7b8e67f2232c26af57c205aaad29fe17754f793404f59c8a730c7a191a/numpy-1.21.6.zip (from https://pypi.org/simple/numpy/) (requires-python:>=3.7,<3.11)
Collecting numpy>=1.10
  Downloading numpy-1.24.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.3/17.3 MB 63.1 MB/s eta 0:00:00
Collecting scikit-learn>=0.16
  Downloading scikit_learn-1.2.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (9.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.6/9.6 MB 79.2 MB/s eta 0:00:00
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.10'): https://files.pythonhosted.org/packages/99/f1/c00d6be56e1a718a3068079e3ec8ce044d7179345280f6a3f5066068af0d/scipy-1.6.2.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.10)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.10'): https://files.pythonhosted.org/packages/fe/fd/8704c7b7b34cdac850485e638346025ca57c5a859934b9aa1be5399b33b7/scipy-1.6.3.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.10)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.10'): https://files.pythonhosted.org/packages/8e/e1/ba481fb2154410f9c5a2d875695873597a7b1bea8c7f87cf444479c3260c/scipy-1.7.0rc1.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.10)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.10'): https://files.pythonhosted.org/packages/a2/ae/ef8b23f5383403a597f2f79df44124a9b8f011a8beb1bca4c6b663a61d15/scipy-1.7.0rc2.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.10)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.10'): https://files.pythonhosted.org/packages/bb/bb/944f559d554df6c9adf037aa9fc982a9706ee0e96c0d5beac701cb158900/scipy-1.7.0.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.10)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.10'): https://files.pythonhosted.org/packages/47/33/a24aec22b7be7fdb10ec117a95e1e4099890d8bbc6646902f443fc7719d1/scipy-1.7.1.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.10)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/0e/23/58c4f995475a2a97cb5f4a032aedaf881ad87cd976a7180c55118d105a1d/scipy-1.7.2.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.7,<3.11'): https://files.pythonhosted.org/packages/61/67/1a654b96309c991762ee9bc39c363fc618076b155fe52d295211cf2536c7/scipy-1.7.3.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.7,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.8,<3.11'): https://files.pythonhosted.org/packages/c0/ad/e3c052ed4e0027a8abef0a5e8441a044427d252d17d9aee06d56e62fc698/scipy-1.8.0rc1.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.8,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.8,<3.11'): https://files.pythonhosted.org/packages/29/d2/151a54944b333e465f98804dced31dab1284f3c37b752b9cefa710b64681/scipy-1.8.0rc2.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.8,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.8,<3.11'): https://files.pythonhosted.org/packages/e4/26/83dd1c6378513a6241d984bda9f08c512b6e35fff13fba3acc1b3c195f02/scipy-1.8.0rc3.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.8,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.8,<3.11'): https://files.pythonhosted.org/packages/22/78/056cc43e7737811b6f50886788a940f852773dd9804f5365952805db9648/scipy-1.8.0rc4.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.8,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.8,<3.11'): https://files.pythonhosted.org/packages/b4/a2/4faa34bf0cdbefd5c706625f1234987795f368eb4e97bde9d6f46860843e/scipy-1.8.0.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.8,<3.11)
  Link requires a different Python (3.11.2 not in: '>=3.8,<3.11'): https://files.pythonhosted.org/packages/26/b5/9330f004b9a3b2b6a31f59f46f1617ce9ca15c0e7fe64288c20385a05c9d/scipy-1.8.1.tar.gz (from https://pypi.org/simple/scipy/) (requires-python:>=3.8,<3.11)
Collecting scipy>=0.19
  Downloading scipy-1.10.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 34.1/34.1 MB 58.8 MB/s eta 0:00:00
Collecting joblib>=1.1.1
  Downloading joblib-1.2.0-py3-none-any.whl (297 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 298.0/298.0 kB 43.1 MB/s eta 0:00:00
Collecting threadpoolctl>=2.0.0
  Downloading threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Building wheels for collected packages: hmmlearn
  Running command python setup.py bdist_wheel
  /usr/local/lib/python3.11/site-packages/setuptools/config/setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
    warnings.warn(msg, warning_class)
  /usr/local/lib/python3.11/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-cpython-311
  creating build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/base.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/stats.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/_utils.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/_version.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/utils.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/__init__.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  creating build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_multinomial_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_poisson_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_base.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_utils.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/conftest.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_categorical_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gmm_hmm_new.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/__init__.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gmm_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gmm_hmm_multisequence.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gaussian_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  running build_ext
  building 'hmmlearn._hmmc' extension
  creating build/temp.linux-x86_64-cpython-311
  creating build/temp.linux-x86_64-cpython-311/src
  gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include -I/usr/local/include/python3.11 -c src/_hmmc.cpp -o build/temp.linux-x86_64-cpython-311/src/_hmmc.o -fvisibility=hidden -g0 -std=c++11
  src/_hmmc.cpp: In function ‘std::tuple<double, pybind11::array_t<double, 16>, pybind11::array_t<double, 16> > forward_scaling(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:66:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
     66 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘std::tuple<double, pybind11::array_t<double, 16> > forward_log(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:120:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    120 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> backward_scaling(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:154:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    154 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> backward_log(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:188:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    188 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> compute_scaling_xi_sum(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:222:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    222 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> compute_log_xi_sum(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:258:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    258 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘std::tuple<double, pybind11::array_t<long int, 16> > viterbi(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:293:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    293 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  error: command '/usr/bin/gcc' failed with exit code 1
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/local/bin/python -u -c '
  exec(compile('"'"''"'"''"'"'
  # This is <pip-setuptools-caller> -- a caller that pip uses to run setup.py
  #
  # - It imports setuptools before invoking setup.py, to enable projects that directly
  #   import from `distutils.core` to work with newer packaging standards.
  # - It provides a clear error message when setuptools is not installed.
  # - It sets `sys.argv[0]` to the underlying `setup.py`, when invoking `setup.py` so
  #   setuptools doesn'"'"'t think the script is `-c`. This avoids the following warning:
  #     manifest_maker: standard file '"'"'-c'"'"' not found".
  # - It generates a shim setup.py, for handling setup.cfg-only projects.
  import os, sys, tokenize
  
  try:
      import setuptools
  except ImportError as error:
      print(
          "ERROR: Can not execute `setup.py` since setuptools is not available in "
          "the build environment.",
          file=sys.stderr,
      )
      sys.exit(1)
  
  __file__ = %r
  sys.argv[0] = __file__
  
  if os.path.exists(__file__):
      filename = __file__
      with tokenize.open(__file__) as f:
          setup_py_code = f.read()
  else:
      filename = "<auto-generated setuptools caller>"
      setup_py_code = "from setuptools import setup; setup()"
  
  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'/tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' bdist_wheel -d /tmp/pip-wheel-9x0tjd2q
  cwd: /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/
  Building wheel for hmmlearn (setup.py) ... error
  ERROR: Failed building wheel for hmmlearn
  Running setup.py clean for hmmlearn
  Running command python setup.py clean
  /usr/local/lib/python3.11/site-packages/setuptools/config/setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
    warnings.warn(msg, warning_class)
  /usr/local/lib/python3.11/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  running clean
  removing 'build/temp.linux-x86_64-cpython-311' (and everything under it)
  removing 'build/lib.linux-x86_64-cpython-311' (and everything under it)
  'build/bdist.linux-x86_64' does not exist -- can't clean it
  'build/scripts-3.11' does not exist -- can't clean it
  removing 'build'
Failed to build hmmlearn
Installing collected packages: threadpoolctl, numpy, joblib, scipy, scikit-learn, hmmlearn
  changing mode of /usr/local/bin/f2py to 755
  changing mode of /usr/local/bin/f2py3 to 755
  changing mode of /usr/local/bin/f2py3.11 to 755
  Running command Running setup.py install for hmmlearn
  /usr/local/lib/python3.11/site-packages/setuptools/config/setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
    warnings.warn(msg, warning_class)
  /usr/local/lib/python3.11/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  running install
  /usr/local/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
    warnings.warn(
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-cpython-311
  creating build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/base.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/stats.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/_utils.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/_version.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/utils.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  copying lib/hmmlearn/__init__.py -> build/lib.linux-x86_64-cpython-311/hmmlearn
  creating build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_multinomial_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_poisson_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_base.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_utils.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/conftest.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_categorical_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gmm_hmm_new.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/__init__.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gmm_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gmm_hmm_multisequence.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  copying lib/hmmlearn/tests/test_gaussian_hmm.py -> build/lib.linux-x86_64-cpython-311/hmmlearn/tests
  running build_ext
  building 'hmmlearn._hmmc' extension
  creating build/temp.linux-x86_64-cpython-311
  creating build/temp.linux-x86_64-cpython-311/src
  gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include -I/usr/local/include/python3.11 -c src/_hmmc.cpp -o build/temp.linux-x86_64-cpython-311/src/_hmmc.o -fvisibility=hidden -g0 -std=c++11
  src/_hmmc.cpp: In function ‘std::tuple<double, pybind11::array_t<double, 16>, pybind11::array_t<double, 16> > forward_scaling(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:66:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
     66 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘std::tuple<double, pybind11::array_t<double, 16> > forward_log(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:120:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    120 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> backward_scaling(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:154:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    154 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> backward_log(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:188:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    188 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> compute_scaling_xi_sum(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:222:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    222 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘pybind11::array_t<double> compute_log_xi_sum(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:258:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    258 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  src/_hmmc.cpp: In function ‘std::tuple<double, pybind11::array_t<long int, 16> > viterbi(pybind11::array_t<double>, pybind11::array_t<double>, pybind11::array_t<double>)’:
  src/_hmmc.cpp:293:39: error: use of deleted function ‘pybind11::gil_scoped_release::gil_scoped_release(const pybind11::gil_scoped_release&)’
    293 |   auto nogil = py::gil_scoped_release{};
        |                                       ^
  In file included from /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/pybind11.h:16,
                   from src/_hmmc.cpp:1:
  /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/.eggs/pybind11-2.10.4-py3.11.egg/pybind11/include/pybind11/gil.h:155:5: note: declared here
    155 |     gil_scoped_release(const gil_scoped_release &) = delete;
        |     ^~~~~~~~~~~~~~~~~~
  error: command '/usr/bin/gcc' failed with exit code 1
  error: subprocess-exited-with-error
  
  × Running setup.py install for hmmlearn did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/local/bin/python -u -c '
  exec(compile('"'"''"'"''"'"'
  # This is <pip-setuptools-caller> -- a caller that pip uses to run setup.py
  #
  # - It imports setuptools before invoking setup.py, to enable projects that directly
  #   import from `distutils.core` to work with newer packaging standards.
  # - It provides a clear error message when setuptools is not installed.
  # - It sets `sys.argv[0]` to the underlying `setup.py`, when invoking `setup.py` so
  #   setuptools doesn'"'"'t think the script is `-c`. This avoids the following warning:
  #     manifest_maker: standard file '"'"'-c'"'"' not found".
  # - It generates a shim setup.py, for handling setup.cfg-only projects.
  import os, sys, tokenize
  
  try:
      import setuptools
  except ImportError as error:
      print(
          "ERROR: Can not execute `setup.py` since setuptools is not available in "
          "the build environment.",
          file=sys.stderr,
      )
      sys.exit(1)
  
  __file__ = %r
  sys.argv[0] = __file__
  
  if os.path.exists(__file__):
      filename = __file__
      with tokenize.open(__file__) as f:
          setup_py_code = f.read()
  else:
      filename = "<auto-generated setuptools caller>"
      setup_py_code = "from setuptools import setup; setup()"
  
  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'/tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' install --record /tmp/pip-record-mbgpavzf/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.11/hmmlearn
  cwd: /tmp/pip-install-__8utuq6/hmmlearn_0fc7076b11a44d018da0491620581d43/
  Running setup.py install for hmmlearn ... error
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> hmmlearn

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip

```

--- a/src/_hmmc.cpp
+++ b/src/_hmmc.cpp
@@ -63,7 +63,7 @@ std::tuple<double, py::array_t<double>, py::array_t<double>> forward_scaling(
   auto scaling_ = py::array_t<double>{{ns}};
   auto scaling = scaling_.mutable_unchecked<1>();
   auto log_prob = 0.;
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   std::fill_n(fwd.mutable_data(0, 0), fwd.size(), 0);
   for (auto i = 0; i < nc; ++i) {
     fwd(0, i) = startprob(i) * frameprob(0, i);
@@ -117,7 +117,7 @@ std::tuple<double, py::array_t<double>> forward_log(
   auto buf = std::vector<double>(nc);
   auto fwdlattice_ = py::array_t<double>{{ns, nc}};
   auto fwd = fwdlattice_.mutable_unchecked<2>();
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   for (auto i = 0; i < nc; ++i) {
     fwd(0, i) = log_startprob(i) + log_frameprob(0, i);
   }
@@ -151,7 +151,7 @@ py::array_t<double> backward_scaling(
   }
   auto bwdlattice_ = py::array_t<double>{{ns, nc}};
   auto bwd = bwdlattice_.mutable_unchecked<2>();
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   std::fill_n(bwd.mutable_data(0, 0), bwd.size(), 0);
   for (auto i = 0; i < nc; ++i) {
     bwd(ns - 1, i) = scaling(ns - 1);
@@ -185,7 +185,7 @@ py::array_t<double> backward_log(
   auto buf = std::vector<double>(nc);
   auto bwdlattice_ = py::array_t<double>{{ns, nc}};
   auto bwd = bwdlattice_.mutable_unchecked<2>();
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   for (auto i = 0; i < nc; ++i) {
     bwd(ns - 1, i) = 0;
   }
@@ -219,7 +219,7 @@ py::array_t<double> compute_scaling_xi_sum(
   auto xi_sum_ = py::array_t<double>{{nc, nc}};
   auto xi_sum = xi_sum_.mutable_unchecked<2>();
   std::fill_n(xi_sum.mutable_data(0, 0), xi_sum.size(), 0);
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   for (auto t = 0; t < ns - 1; ++t) {
     for (auto i = 0; i < nc; ++i) {
       for (auto j = 0; j < nc; ++j) {
@@ -255,7 +255,7 @@ py::array_t<double> compute_log_xi_sum(
   auto log_xi_sum = log_xi_sum_.mutable_unchecked<2>();
   std::fill_n(log_xi_sum.mutable_data(0, 0), log_xi_sum.size(),
               -std::numeric_limits<double>::infinity());
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   for (auto t = 0; t < ns - 1; ++t) {
     for (auto i = 0; i < nc; ++i) {
       for (auto j = 0; j < nc; ++j) {
@@ -290,7 +290,7 @@ std::tuple<double, py::array_t<ssize_t>> viterbi(
   auto viterbi_lattice_ = py::array_t<double>{{ns, nc}};
   auto state_sequence = state_sequence_.mutable_unchecked<1>();
   auto viterbi_lattice = viterbi_lattice_.mutable_unchecked<2>();
-  auto nogil = py::gil_scoped_release{};
+  py::gil_scoped_release nogil;
   for (auto i = 0; i < nc; ++i) {
     viterbi_lattice(0, i) = log_startprob(i) + log_frameprob(0, i);
   }

